### DISCUSSION TITLE
500多个节点就卡得不行

### DISCUSSION THREAD

**Discussion by zmcode** [antvis/X6#2460] [Q&A]
![image](https://user-images.githubusercontent.com/48686959/182787596-c0d4abdc-354c-4aec-a5e0-cd41b9d649f5.png)

有什么办法优化吗?

**Comment by wb-wenbei** [comment]
参考一下
[前端图可视化引擎antv的g6和x6区别是什么，如何选择？](https://www.zhihu.com/question/435855401/answer/1639313761)

**Reply by NewByVector** [reply]
@yaojin2070 500 个这种简单节点不应该这么卡的，有 demo 看下吗？

**Reply by zmcode** [reply]
> @yaojin2070 500 个这种简单节点不应该这么卡的，有 demo 看下吗？

公司的代码 不好发, 大概是600个节点, 渲染是4-5秒, 前期我要计算每个图片节点的大小, 所以需要插入一次dom, 获取每个节点的 大小, 再设置对应节点的width 和height,  如果上千个就更加,慢了, 之前还需要移动到节点, 对应关联的节点高亮的, 因为我是使用html的shape, 我使用dom直接查询每个节点, 来使节点高亮, 其他不关联的不高亮, 但是边我无法通过dom的形式来改变它的透明度, 如果使用setAttr, 非常卡, 一次会导致很多边的节的节点重新渲染, 哪怕是仅仅操作dom, 上千节点的时候也会很卡, 可是我的业务需求又要编辑又要展示,我都不知道该用x6还是g6好

**Reply by qiufeihong2018** [reply]
@zmcode 请问你解决了吗？

**Reply by zmcode** [reply]
卡是因为使用的是斥力布局, 不断改变位置来操作dom,模拟动画效果
 我当时记得源码内部使用的d3的斥力, d3内部有一个tick函数, tick直接跳过动态变化位置, 这样就会避免卡顿了, 但是就没有动画效果了

**Comment by zmcode** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。

**Comment by ghost** [comment]
底层是SVG，参考SVG 一般优化方法
